root/static/scripts/relationship-editor/components/RelationshipPhraseGroup.js

```diff
@@ -143,7 +143,11 @@ const RelationshipPhraseGroup = (React.memo<PropsT>(({
       [backward ? 'entity1' : 'entity0']: source,
       attributes: newAttributesData,
       linkOrder: maxLinkOrder > 0 ? (maxLinkOrder + 1) : 0,
-      linkTypeID: linkTypeId,
+      /*
+       * The `typeId` on `RelationshipLinkTypeGroupT` stores null types as 0
+       * for easier sorting. Covnert them back to null here.
+       */
+      linkTypeID: linkTypeId || null,
```

Comment:
Yes, 0 is falsey.  It seems pretty clear to me. `x || 'fallback'` is a ubiquitous pattern in JS.

Reply:
Sure, it's just `linkTypeId` is a number, but we're using the number as a falsey boolean. But I guess the argument is that 0 is never a valid ID anyway so it's safe to treat it as just false? :)

Reply:
I guess in any case the main issue I'd have here is that it's weird to turn null into 0 and then back into null because 0 is not null, and that makes everything confusing, lol. But probably not an issue that should block merging this improvement.

Reply:
Would [adding this parenthesis](https://github.com/metabrainz/musicbrainz-server/pull/2868/files#r1118966547) make it more clear?

Reply:
Maybe, but it's probably not needed anyway. I'm surprised flow doesn't complain about the number vs boolean thing in the same way it complains about strings (hence all our using nonEmpty all around) but it's fine.

Reply:
Well, 0 isn't valid on `RelationshipStateT` at least (or any falsey value except `null`, so even if `linkTypeId` were `''` or `undefined`, the `||` would still be valid).  The only reason we store it as 0 elsewhere is 'cause we sort the sections by link type ID a lot, and it makes it easier to just do `a.typeId - b.typeId` in the sort functions.

I'm not sure `linkTypeID: linkTypeId === 0 ? null : linkTypeId,` is clearer to me -- I kinda find it harder to parse a ternary than a `|| null` fallback, but if you prefer the explicitness of the former I could change it.

Reply:
It was about the parens :) But if @mwiencek is ok with those, then I think they're fine.